tcms/testruns/views.py

```diff
@@ -63,55 +61,58 @@ def post(self, request):
                                                'category',
                                                'priority').order_by('pk')
 
-        if request.POST.get('POSTING_TO_CREATE'):
-            form = NewRunForm(request.POST)
-            form.populate(product_id=test_plan.product_id)
-
-            if form.is_valid():
-                # Process the data in form.cleaned_data
-                default_tester = form.cleaned_data['default_tester']
-
-                test_run = TestRun.objects.create(
-                    product_version=test_plan.product_version,
-                    stop_date=None,
-                    summary=form.cleaned_data.get('summary'),
-                    notes=form.cleaned_data.get('notes'),
-                    plan=test_plan,
-                    build=form.cleaned_data['build'],
-                    manager=form.cleaned_data['manager'],
-                    default_tester=default_tester,
-                )
-
-                loop = 1
-                for case in form.cleaned_data['case']:
-                    try:
-                        tcp = TestCasePlan.objects.get(plan=test_plan, case=case)
-                        sortkey = tcp.sortkey
-                    except ObjectDoesNotExist:
-                        sortkey = loop * 10
-
-                    test_run.add_case_run(case=case, sortkey=sortkey,
-                                          assignee=default_tester)
-                    loop += 1
-
-                return HttpResponseRedirect(
-                    reverse('testruns-get', args=[test_run.pk, ])
-                )
+        form = NewRunForm(initial={
+            'summary': 'Test run for %s' % test_plan.name,
+            'manager': test_plan.author.email,
+            'default_tester': request.user.email,
+            'notes': '',
+            'plan': plan_id
+        })
+        form.populate(product_id=test_plan.product_id)
 
-        else:
-            form = NewRunForm(initial={
-                'summary': 'Test run for %s' % test_plan.name,
-                'manager': test_plan.author.email,
-                'default_tester': request.user.email,
-                'notes': '',
-            })
-            form.populate(product_id=test_plan.product_id)
+        context_data = {
+            'test_plan': test_plan,
+            'test_cases': tcs_values,
+            'form': form,
+            'disabled_cases': get_disabled_test_cases_count(test_cases)
+        }
+        return render(request, self.template_name, context_data)
 
+    def post(self, request):
+        form = NewRunForm(data=request.POST)
+
+        if form.is_valid():
```

Comment:
There was a previous comment about `form.populate()` which seems to be missing now. Can you elaborate more on that ?

Reply:
I thought I wont need it. But now I see that if the form fails I have to populate it, so the dropdown with the products will show correct values.